scripts.js

```diff
@@ -1,348 +1,384 @@
 /** Generates solution urls for GitHub. */
-function gitHubUrls({ user, repo, branch = "main", path }) {
-  return {
-    solutionUrl: (day, part) => `https://raw.githubusercontent.com/${user}/${repo}/${branch}/${path(day, part)}`,
-    solutionWebUrl: (day, part) => `https://github.com/${user}/${repo}/blob/${branch}/${path(day, part)}`,
-  };
+function gitHubUrls({user, repo, branch = (year => "main"), path}) {
+	return {
+		solutionUrl: (year, day, part) => `https://raw.githubusercontent.com/${user}/${repo(year)}/${branch(year)}/${path(year, day, part)}`,
+		solutionWebUrl: (year, day, part) => `https://github.com/${user}/${repo(year)}/blob/${branch(year)}/${path(year, day, part)}`,
+	};
 }
 
 /** Pads the given number with the given number of zeros. */
 function pad(n, zeros) {
-  return `${n}`.padStart(zeros, "0");
+	return `${n}`.padStart(zeros, "0");
 }
 
 window.addEventListener("load", async () => {
-  const year = 2022;
-  const shortYear = year % 100;
 
-  const fwcdPaths = await (await fetch(`https://raw.githubusercontent.com/fwcd/advent-of-code-${year}/main/paths.json`)).json();
-  const estgPaths = await (await fetch(`https://raw.githubusercontent.com/estugon/advent-of-code-${year}/main/paths.json`)).json();
+	const fwcdPaths = async year => {
+		return await (await fetch(`https://raw.githubusercontent.com/fwcd/advent-of-code-${year}/main/paths.json`)).json();
+	}
+	const estgPaths = async year => {
+		return await (await fetch(`https://raw.githubusercontent.com/estugon/advent-of-code-${year}/main/paths.json`)).json();
```

Comment:
Du musst den Rückgabewert in einer Async Funktion nicht `await`en.

Reply:
Hab ich von vorher übernommen, aber nehm ich gerne raus.